libraries/botbuilder-dialogs-adaptive/src/actions/httpRequest.ts

```diff
@@ -286,79 +287,113 @@ export class HttpRequest<O extends object = {}> extends Dialog<O> implements Htt
+                case HttpMethod.PATCH:
+                case HttpMethod.POST:
+                    response = await fetch(instanceUrl, {
+                        method: instanceMethod,
+                        headers: instanceHeaders,
+                        body: instanceBody,
+                    });
+                    break;
+            }
 
-        const result = new Result(response.headers);
-        result.statusCode = response.status;
-        result.reasonPhrase = response.statusText;
-
-        switch (this.responseType.getValue(dc.state)) {
-            case ResponsesTypes.Activity:
-                result.content = await response.json();
-                dc.context.sendActivity(result.content as Activity);
-                break;
-            case ResponsesTypes.Activities:
-                result.content = await response.json();
-                dc.context.sendActivities(result.content as Activity[]);
-                break;
-            case ResponsesTypes.Json: {
-                const content = await response.text();
-                try {
-                    result.content = JSON.parse(content);
-                } catch {
-                    result.content = content;
+            const result = new Result(response.headers);
+            result.statusCode = response.status;
+            result.reasonPhrase = response.statusText;
+
+            switch (this.responseType.getValue(dc.state)) {
+                case ResponsesTypes.Activity:
+                    result.content = await response.json();
+                    dc.context.sendActivity(result.content as Activity);
+                    break;
+                case ResponsesTypes.Activities:
+                    result.content = await response.json();
+                    dc.context.sendActivities(result.content as Activity[]);
+                    break;
+                case ResponsesTypes.Json: {
+                    const content = await response.text();
+                    try {
+                        result.content = JSON.parse(content);
+                    } catch {
+                        result.content = content;
+                    }
+                    break;
                 }
-                break;
+                case ResponsesTypes.Binary: {
+                    const buffer = await response.arrayBuffer();
+                    result.content = new Uint8Array(buffer);
+                    break;
+                }
+                case ResponsesTypes.None:
+                default:
+                    break;
             }
-            case ResponsesTypes.Binary: {
-                const buffer = await response.arrayBuffer();
-                result.content = new Uint8Array(buffer);
-                break;
+
+            traceInfo.response = result;
+
+            await this.EndDialogWithResultAsync(dc, result, traceInfo);
+        } catch (err) {
+            const result = new Result();
+            result.content = err.message;
```

Comment:
Move these lines inside the if

Reply:
I think I need to leave the line 368 out so I can use result outside the if

Reply:
Why do you need it outside?
```
} catch (err) {
      if (err instanceof FetchError) {
           const result = new Result();
           result.statusCode = StatusCodes.NOT_FOUND;
           result.content = err.message;

           await this.endDialogWithResult(dc, result, traceInfo);
      } else {
           throw err;
      }
}
```

Reply:
Now that I see it, I needed it for a traceInfo line, but I remove it.